features/202-session-management/api.md

```diff
@@ -0,0 +1,100 @@
+# Session Management APIs
+
+## SDK API
+```typescript
+interface UserAgent {
+    raw: string;
+    name: string; // browser name or app ID
+    version: string; // browser version or app version
+    os: string;
+    osVersion: string;
+    device: string;
+}
+
+interface Session {
+    id: string;
+    identityID: string;
+    createdAt: Date;
+    lastAccessedAt: Date;
+    createdByIP: string;
+    lastAccessedIP: string;
+    createdByUserAgent: UserAgent;
+    lastAccessedUserAgent: UserAgent;
+    name: string;
+    data: object;
+}
```

Comment:
Consider the most common use case of rendering a Session to User's interface, usually it make more sense to display Location instead of IP addresses.

So our session data should include a list of Locations that currently used by the sessions / a few last locations used historically? 

So I think we should add an attributed call something like `lastAccessedLocation` in session? And for now we get the location from IP address.

And we should remember (open an issue?) to add related APIs for people to queries a list of locations when we implement the logging of session data feature (Which feature issue it is?)

Reply:
I'd prefer not to include it for now, since there are some points to consider:
- Format of location (just country / more precise location)
- Geo IP database customization
- Localization
- Privacy & compliance

Reply:
But if we don't have it now, we almost defeated the MVP purpose of this feature issue.... :( I won't consider a session list that don't show sessions country good enough to let me check if I have unwanted session....

Reply:
I would consider these points are essential to including location information, and changing it after release would lead to backward incompatibility. We can open another issue to address it before official release?

For now, since we provide IP, app developer can solve and implement these points themselves, and our management portal can also implemented it ourselves.

Reply:
Sure. Please open another feature issue than. Thanks.

Reply:
opened #354 to track